signer/src/transaction_coordinator.rs

```diff
@@ -169,6 +177,12 @@ where
             .await?
             .ok_or(Error::NoChainTip)?;
 
+        if self.needs_dkg(&bitcoin_chain_tip).await? == DkgState::NeedsDkg {
```

Comment:
Since this is not guarded by some `self.is_coordinator` condition, do we risk to have concurrent DKGs?

Reply:
Oh, I just forgot to add that in.